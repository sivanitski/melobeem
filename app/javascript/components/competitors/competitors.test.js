import Enzyme, { mount } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import React from "react";
import renderer from "react-test-renderer";

import Competitors from "./competitors";

const mockProps = {
  competitors: [
    {
      id: `111`,
      name: `Julia Mur`,
      avatar: `some/url/avatar.png`,
      likes: 1000,
      parentName: `John Depp`,
      rank: 2,
    },
    {
      id: `112`,
      name: `Elena Black`,
      avatar: `some/url/avatar.png`,
      likes: 2000,
      parentName: `White Black`,
      rank: 1,
    },
    {
      id: `113`,
      name: `Tom Raddle`,
      avatar: `some/url/avatar.png`,
      likes: 100,
      parentName: `Some Name`,
      rank: 1,
    },
  ],
};

Enzyme.configure({
  adapter: new Adapter(),
});

describe("Competitors component", () => {
  it("should render Competitors component", () => {
    const component = renderer
      .create(<Competitors competitors={mockProps.competitors} />)
      .toJSON();

    expect(component).toMatchSnapshot();
  });
});

describe("Competitors handlers", () => {
  it("should make  slider click ", () => {
    const component = mount(
      <Competitors competitors={mockProps.competitors} />
    );

    const competitorsSwiperMenu = component.find(`competitors__menu`);
    console.log(competitorsSwiperMenu);
  });
});

// describe("Button component", () => {
//   it("should call onClick method", () => {
//     const onSliderClick = jest.fn();
//     const component = mount(<Competitors competitors={mockProps.competitors} />);
//     expect(mockCallBack.mock.calls.length).toBe(0);
//     component.find(".btn").simulate("click");
//     expect(mockCallBack.mock.calls.length).toBe(1);
//   });
// });
