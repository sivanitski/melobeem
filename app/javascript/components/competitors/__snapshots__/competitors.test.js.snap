// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Competitors component should render Competitors component 1`] = `
<div
  className="competitors"
>
  <div
    className="competitors__title headline--medium"
  >
    Leaderboard 
  </div>
  <div
    className="swiper-container competitors__swiper swiper"
  >
    <div
      className="swiper-wrapper"
    >
      <div
        className="swiper-slide competitors__swiper__item headline active"
        onClick={[Function]}
      >
        1
      </div>
      <div
        className="swiper-slide competitors__swiper__item headline text-pink"
        onClick={[Function]}
      >
        2
      </div>
      <div
        className="swiper-slide competitors__swiper__item headline text-pink"
        onClick={[Function]}
      >
        3
      </div>
      <div
        className="swiper-slide competitors__swiper__item headline text-pink"
        onClick={[Function]}
      >
        4
      </div>
      <div
        className="swiper-slide competitors__swiper__item headline text-pink"
        onClick={[Function]}
      >
        5
      </div>
      <div
        className="swiper-slide competitors__swiper__item headline text-pink"
        onClick={[Function]}
      >
        6
      </div>
    </div>
  </div>
  <div
    className="competitors__level"
  >
    <div
      className="competitors__level__amount"
    >
      Level 
      1
    </div>
    <div
      className="competitors__level__comment text-grey"
    >
      (From 
      0
       to
       
      5
       Votes)
       
    </div>
  </div>
</div>
`;
